From bf2dacc8657217f4715afb1bf8bb9f3d3e7772e2 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Thu, 16 Nov 2023 06:57:30 +0100
Subject: [PATCH 0039/1695] plat-stm32mp1: scmi_server: report invalid
 regulator state request

Changes the SCMI return code from SCMI_GENERIC_ERROR to
SCMI_INVALID_PARAMETERS when the requested state is not one of
the 2 supported SCMI voltage domain states
(SCMI_VOLTAGE_DOMAIN_CONFIG_ARCH_ON or
SCMI_VOLTAGE_DOMAIN_CONFIG_ARCH_OFF).

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 966568bd5c06c2700cfe34b0cd51af15aa5d6a20 plat-stm32mp1: scmi_server: report invalid regulator state request]
Change-Id: I02627926f6bda2a6d3db03e317aa5a15b39309e7
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
[etienne: cherry-picked from 305e38d93acfeabdb33cd014252f68f25a9be2cf]
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
---
 core/arch/arm/plat-stm32mp1/scmi_server.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/arch/arm/plat-stm32mp1/scmi_server.c b/core/arch/arm/plat-stm32mp1/scmi_server.c
index 36c1b8e0e..fee31ff6c 100644
--- a/core/arch/arm/plat-stm32mp1/scmi_server.c
+++ b/core/arch/arm/plat-stm32mp1/scmi_server.c
@@ -893,7 +893,7 @@ int32_t plat_scmi_voltd_set_config(unsigned int channel_id,
 			}
 			break;
 		default:
-			return SCMI_GENERIC_ERROR;
+			return SCMI_INVALID_PARAMETERS;
 		}
 
 		return SCMI_SUCCESS;
-- 
2.47.2

